pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1779,6 +1811,15 @@ func (c *Cloud) CreateDisk(volumeOptions *VolumeOptions) (KubernetesVolumeID, er
 	return volumeName, nil
 }
 
+// checkEncryptionKey tests that given encryption key exists.
+func (c *Cloud) checkEncryptionKey(keyId string) error {
+	input := &kms.DescribeKeyInput{
+		KeyId: aws.String(keyId),
+	}
+	_, err := c.kms.DescribeKey(input)
```

Comment:
So in theory at least this could fail for other reasons.  The most notable one is lack of permissions (and I'll touch on that separately).  Can we check the actual error code and wrap it, like e.g. https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/aws.go#L1457-L1463

I believe the error code should be "NotFoundException" (based on `aws --debug kms describe-key --key-id nope`)

It's probably a good idea to wrap the AWS error in both cases, as the raw AWS errors can be a bit cryptic.

i.e.

if code == "NotFoundException" {
return fmt.Errorf("KMS key %q not found: %v", keyId, err)
} else {
return fmt.Errorf("error querying for KMS key %q: %v", keyId, err)
}